tez-runtime-library/src/main/java/org/apache/tez/runtime/library/common/sort/impl/dflt/DefaultSorter.java

```diff
@@ -1399,4 +1409,18 @@ private void mergeParts() throws IOException, InterruptedException {
       }
     }
   }
+
+  /**
+   * Increments numSpills local counter by taking into consideration
+   * the max limit on spill files being generated by the job.
+   * If limit is reached, this function throws an IOException
+   */
+  private void incrementNumSpills() throws IOException {
```

Comment:
shouldn't this duplicated logic go to ExternalSorter instead of all children (PipelinedSorter, DefaultSorter)?